Neos.Flow/Classes/Mvc/Routing/RouteConfiguration.php

```diff
@@ -0,0 +1,120 @@
+<?php
+namespace Neos\Flow\Mvc\Routing;
+
+/*
+ * This file is part of the Neos.Flow package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\Flow\Annotations as Flow;
+use Neos\Flow\Configuration\ConfigurationManager;
+use Neos\Flow\Mvc\Exception\InvalidRouteSetupException;
+
+/**
+ * The default router configuration
+ *
+ * @Flow\Scope("singleton")
+ */
+final class RouteConfiguration
+{
+    /**
+     * Array of routes to match against
+     *
+     * @var Routes
+     */
+    private Routes $routes;
```

Comment:
Does this work currently?

I think it has to be nullable, since it is not initialized during construction time
```suggestion
    private ?Routes $routes = null;
```

Reply:
btw: you can omit the `routesCreated` field and just replace the check in line 51 with
```php
if ($this->routes === null) {
    $this->routes = $this->createRoutesFromConfiguration();
}

Reply:
fun fact it doesnt have to be null able (as long as you use isset() or `??` to check on it)

Reply:
True, it could even be simplified to a oneliner
https://3v4l.org/JBB4r#v8.1.9

Reply:
exactly what i suggested here: https://github.com/neos/flow-development-collection/pull/2970#discussion_r1124256695 :P